File: back/Instituicao.py
class Instituicao:
    def __init__(self, db):
        self.db = db

    def add_instituicao(self, nome_instituicao, cnpj):
        try:
            self.db.cursor.execute(
                "INSERT INTO instituicao (Nome, Cnpj) VALUES (%s, %s)", 
                (nome_instituicao, cnpj)
            )
            self.db.conn.commit()
            return self.db.cursor.lastrowid
        except Exception as e:
            print(f"Erro ao adicionar instituição ao banco de dados: {e}")
            raise e
        
    def get_instituicoes(self):
        try:
            self.db.cursor.execute("SELECT * FROM instituicao ORDER BY Nome")
            result = self.db.cursor.fetchall()
            return result
        except Exception as e:
            print(f"Erro ao obter instituições do banco de dados: {e}")
            raise e

    def instituicao_avaliacao_insert(self, avaliacao_id, instituicao_id):
        try:
            self.db.cursor.execute("UPDATE avaliacao set ID_Instituicao = %s WHERE ID = %s", (instituicao_id, avaliacao_id))
            self.db.conn.commit()
        except Exception as e:
            print(f"Erro ao atualizar instituição na avaliação: {e}")
            raise e

File: backjs/classes/Instituicao.js
class Instituicao {
    constructor(db) {
        this.db = db;
    }

    async addInstituicao(nomeInstituicao, cnpj) {
        const query = "INSERT INTO instituicao (Nome, Cnpj) VALUES (?, ?)";
        try {
            const result = await this.db.executeQuery(query, [nomeInstituicao, cnpj]);
            return result.insertId; // Retorna o ID da nova instituição inserida
        } catch (err) {
            console.error("Erro ao adicionar instituição ao banco de dados:", err);
            throw err;
        }
    }

    async getInstituicoes() {
        const query = "SELECT * FROM instituicao ORDER BY Nome";
        try {
            const instituicoes = await this.db.fetchAll(query);
            return instituicoes;
        } catch (err) {
            console.error("Erro ao obter instituições do banco de dados:", err);
            throw err;
        }
    }

    async instituicaoAvaliacaoInsert(avaliacaoId, instituicaoId) {
        const query = "UPDATE avaliacao SET ID_Instituicao = ? WHERE ID = ?";
        try {
            await this.db.executeQuery(query, [instituicaoId, avaliacaoId]);
        } catch (err) {
            console.error("Erro ao atualizar instituição na avaliação:", err);
            throw err;
        }
    }
}

module.exports = Instituicao;